src/utils/update/trusts_data_fetch.ts

```diff
@@ -0,0 +1,318 @@
+import { ApplicationData } from "../../model";
+import { yesNoResponse } from "../../model/data.types.model";
+import { RoleWithinTrustType } from "../../model/role.within.trust.type.model";
+import { IndividualTrustee, TrustCorporate } from "../../model/trust.model";
+
+export const retrieveTrustsData = (appData: ApplicationData) => {
+  if (!hasFetchedTrustsData(appData)) {
+    initialiseTrustsUpdateAppData(appData);
+    if (appData.update) {
+      retrieveTrusts(appData);
+      appData.update.trust_data_fetched = true;
+    }
+  }
+};
+
+const hasFetchedTrustsData = (appData: ApplicationData) => appData?.update?.trust_data_fetched ?? false;
+
+const initialiseTrustsUpdateAppData = (appData: ApplicationData) => {
+  if (!appData.update) {
+    appData.update = {};
+  }
+
+  appData.update.review_trusts = [];
+};
+
+const retrieveTrusts = (appData: ApplicationData) => {
+  if (!appData.update) {
+    appData.update = {};
+  }
```

Comment:
Both initialiseTrustsUpdateAppData and retrieveTrusts do the same thing. Do we need both?